tests/integration/test_charm.py

```diff
@@ -0,0 +1,276 @@
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.connector import MysqlConnector
+from tests.integration.constants import (
+    DATA_INTEGRATOR,
+    DATABASE_NAME,
+    MONGODB,
+    MYSQL,
+    POSTGRESQL,
+)
+from tests.integration.helpers import (
+    build_postgresql_connection_string,
+    create_table_mysql,
+    fetch_action_get_credentials,
+    insert_data_mysql,
+    read_data_mysql,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    data_integrator_charm = await ops_test.build_charm(".")
+    await asyncio.gather(
+        ops_test.model.deploy(
+            data_integrator_charm, application_name="data-integrator", num_units=1, series="jammy"
+        )
+    )
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR])
+    assert ops_test.model.applications[DATA_INTEGRATOR].status == "blocked"
+
+    # config database name
+
+    config = {"database": DATABASE_NAME}
+    await ops_test.model.applications[DATA_INTEGRATOR].set_config(config)
+
+    # test the active/waiting status for relation
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR])
+    assert ops_test.model.applications[DATA_INTEGRATOR].status == "active"
+
+
+@pytest.mark.skip  # skipping as we can't deploy MYSQL (https://github.com/canonical/mysql-operator/pull/73)
+async def test_deploy_and_relate_mysql(ops_test: OpsTest):
+    """Test the relation with MySQL and database accessibility."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            "mysql", channel="edge", application_name=MYSQL, num_units=1, series="focal"
+        )
+    )
+    await ops_test.model.wait_for_idle(apps=[MYSQL])
+    assert ops_test.model.applications[MYSQL].status == "active"
+    await ops_test.model.add_relation(DATA_INTEGRATOR, MYSQL)
+    await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR, MYSQL])
+    assert ops_test.model.applications[DATA_INTEGRATOR].status == "active"
+
+    # get credential for MYSQL
+    credentials = fetch_action_get_credentials(ops_test.model.applications[DATA_INTEGRATOR][0])
+
+    # test connection for MYSQL with retrieved credentials
+    # connection configuration
+    config = {
+        "user": credentials[MYSQL]["username"],
+        "password": credentials[MYSQL]["password"],
+        "host": credentials[MYSQL]["endpoints"].split(":")[0],
+        "database": DATABASE_NAME,
+        "raise_on_warnings": False,
+    }
+
+    with MysqlConnector(config) as cursor:
+
+        create_table_mysql(cursor, DATABASE_NAME)
+
+        insert_data_mysql(
+            cursor,
+            DATABASE_NAME,
+            credentials[MYSQL]["username"],
+            credentials[MYSQL]["password"],
+            credentials[MYSQL]["endpoints"],
+            credentials[MYSQL]["version"],
+            credentials[MYSQL]["read-only-endpoints"],
+        )
+
+        rows = read_data_mysql(cursor, credentials[MYSQL]["username"])
```

Comment:
Major comment:

I would have splitted the creation and read part from the reading. 

These are indeed two different unit of works. Otherwise we are not really testing persistence here...